community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyStore.java

```diff
@@ -361,9 +361,13 @@ public PropertyRecord getRecord( PropertyRecord record )
         }
     }
 
-    public PageCursor newPageCursor( int mode ) throws IOException
+    public PageCursor newReadCursor( long recordId ) throws IOException
     {
-        return storeFile.io( 0, mode );
+        PageCursor cursor =  storeFile.io( pageIdForRecord( recordId ), PF_SHARED_LOCK );
+        if (!cursor.next())
+            throw new IOException( "Record not found: "+recordId );
```

Comment:
On this level IOException is thrown, which will be converted to an UnderlyingStorageException. It is much more severe than an EntityNotFoundException, and indicates that something is really wrong.